src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -0,0 +1,596 @@
+        }
+
+        int newArgsStartPos = 1;
+        if (2 <= args.length)
+        {
+            final ActionArgumentType actionArgumentType = holder.getActionArgument().getType();
+            final int allowedSpaceCount = actionArgumentType.allowedSpaceCount();
+            int wordCount = 1;
+            final StringBuilder sb = new StringBuilder(args[1]);
+            while ((wordCount <= allowedSpaceCount) && ((wordCount + 1) < args.length))
+            {
+                final String nextPotentialWord = args[1 + wordCount];
+                if (null == possibleActionCommands.get(nextPotentialWord))
+                {
+                    sb.append(' ').append(nextPotentialWord);
+                    ++wordCount;
+                }
+                else
+                {
+                    break;
+                }
+            }
+            final String potentialArgumentValue = sb.toString();
+            newArgsStartPos = 1 + wordCount;
+            final Object parsedObject = actionArgumentType.parse(server, sender, pos, parsedHolders, potentialArgumentValue);
+            if (null == parsedObject)
+            {
+                final List<String> tabCompletions = actionArgumentType.getTabCompletions(server, pos, potentialArgumentValue);
+                return new ParsingResult(tabCompletions, actionMenuTreeNode, parsedActionArgumentList, potentialArgumentValue);
+            }
+            else
+            {
+                final ActionArgument actionArgument = holder.getActionArgument();
+                actionArgument.setValue(parsedObject);
+                parsedHolders.add(holder);
+
+                // add any subArguments
+                final TreeNode<Menu> treeNode = holder.getTreeNode();
+                final List<ActionArgument> subActionArgumentList = actionArgument.getActionArgumentList();
+                for (final ActionArgument subActionArgument : subActionArgumentList)
+                {
+                    possibleActionCommands.put(subActionArgument.getName() + ":", new ActionMenuHolder(treeNode, subActionArgument));
+                }
+
+
+                if (newArgsStartPos == args.length)
+                {
+                    final List<String> tabCompletions = Collections.emptyList();
+                    return new ParsingResult(tabCompletions, actionMenuTreeNode, parsedActionArgumentList, (String) null);
+                }
+            }
+        }
+
+        final String[] newArgs = new String[args.length - newArgsStartPos];
+        System.arraycopy(args, newArgsStartPos, newArgs, 0, newArgs.length);
+        return getTabCompletionsAndParsingHolders(parsedHolders, actionMenuTreeNode, parsedActionArgumentList, possibleActionCommands, server, sender, newArgs, pos);
+    }
+
+    @Override
+    public boolean isUsernameIndex(@NotNull final String[] args, final int index)
+    {
+        super.isUsernameIndex(args, index);
+        // TODO: implement
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)